packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -226,67 +226,14 @@ where
             consensus_update: update,
         })
     }
-}
 
-#[async_trait::async_trait]
-impl<P> TxBuilderService<EthEureka, CosmosSdk> for TxBuilder<P>
-where
-    P: Provider + Clone,
-{
     #[tracing::instrument(skip_all)]
-    async fn relay_events(
+    async fn get_update_headers(
```

Comment:
This diff is super confusing to look at. I essentially moved a lot of the relay_events logic up into a get_update_headers function, which is why this looks so strange.